articles/native-platforms/ios-swift/03-session-handling.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Session Handling
+description: This tutorial will show you how to handle sessions in your app, with the aim of preventing the user from being asked for credentials each time the app is launched.
+---
+
+::: panel-info System Requirements
+This tutorial and seed project have been tested with the following:
+
+* CocoaPods 1.0.0
+* XCode 7.3 (7D175)
+* Simulator - iPhone 6 - iOS 9.3 (13E230)
+  :::
+
+<%= include('../_includes/_package', {
+  pkgRepo: 'native-mobile-samples',
+  pkgBranch: 'master',
+  pkgPath: 'iOS/basic-sample-swift',
+  pkgFilePath: 'iOS/basic-sample-swift/SwiftSample/Info.plist',
+  pkgType: 'replace'
+}) %>
+
+### Before Starting
+
+#### i. Be familiar with Lock Login
+
+This tutorial assumes you're using the Lock library for handling login. Make sure you've integrated this library into your project and you're familiar with it. **If you're not sure, check out [this tutorial](01-login.md) first.**
+
+#### ii. Add the SimpleKeychain dependency
+
+We're going to use the [SimpleKeychain](https://github.com/auth0/SimpleKeychain) library to help us manage user credentials. Make sure you integrate it before proceeding.
+
+##### a. Carthage
+
+If you are using Carthage, add the following line to the `Cartfile`:
+
+```ruby
+github "auth0/SimpleKeychain"
+```
+
+Then, run `carthage bootstrap`.
+
+> For more information about Carthage usage, check [their official documentation](https://github.com/Carthage/Carthage#if-youre-building-for-ios-tvos-or-watchos).
+
+##### b. Cocoapods
+
+If you are using [Cocoapods](https://cocoapods.org/), add these lines to your `Podfile`:
+
+```ruby
+use_frameworks!
+pod 'SimpleKeychain', '~> 0.7'
+```
+
+Then, run `pod install`.
+
+> For further reference on Cocoapods, check [their official documentation](http://guides.cocoapods.org/using/getting-started.html).
+
+### 1. Store the user's idToken
+
+> The `idToken` is a string representing, basically, the user's [JWT token](https://en.wikipedia.org/wiki/JSON_Web_Token).
+
+We will store this `idToken` **upon a successful login**, in order to prevent the user from being asked for login credentials again whenever he re-launches the app.
+
+Once the user has logged in, you get both an `A0Profile` and an `A0Token` object, coming as follows:
+
+```swift
+let controller = A0Lock.sharedLock().newLockViewController()
+controller.onAuthenticationBlock = { profile, token in
+	// do something with profile and token
+}
+```
+
+> Even though `profile` and `token` come as optional values, the [A0LockViewController](https://github.com/auth0/Lock.iOS-OSX/blob/master/Lock/UI/A0LockViewController.h) documentation guarantees that both of them will only be `nil` iff login is disabled after sign up. We assume that's not our case; therefore, we will perform force-unwrap to get their values.
```

Comment:
I guessed as much, but I don't think you should use that kind of 'code expressions' on a coloquial text. Especially in this case where it could be misunderstood as typo. It leads to confusion. Better to say something along the lines of: 

"guarantees that both of them will only be nil when login is disabled"